### DISCUSSION TITLE
[Feature] Preserve (actual) file timestamps when uploading

### DISCUSSION THREAD

**Discussion by parasiteoflife** [immich-app/immich#2581] [Feature Request]
### The feature

When we upload files the dates are shown correctly on Immich but the actual files in the filesystem/hard drive/SSD/whatever are set to the upload date instead of preserving the original modification and accessed times (actually, I'm mostly asking here for modification times to be preserved, accessed times I consider a plus).

This is useful for photos and videos that don't have timestamp metadata embedded and the only date is the filesystem's one, in that case if Immich crashes, corrupts, explodes, or something else happens, we still have the correct dates for those files in the storage drive and don't have to rely on that part on Immich itself.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by auanasgheps** [comment]
This is especially useful when downloading or managing a bunch of photos and videos that were uploaded in Immich. Sorting might be a bit messy.
Please keep original timestamps!

As an example, Davinci Resolve only reads file created and modified date.

**Comment by jrasm91** [comment]
We stream the file over http and write it to disk. It makes sense that a new file results in new timestamps. We do pass the original created/modified fields in the upload request though, so we should be able to use something like [this](https://nodejs.org/api/fs.html#filehandleutimesatime-mtime) to re-set the original timestamps on the newly created file.

**Reply by auanasgheps** [reply]
Syncthing, as an example, preserves create date.

**Comment by nomandera** [comment]
New user; came to complement the project and to make this feature request only to find it exists already. :)

This is the kind of detail that can really make a difference. There are cases where no metadata exists, is corrupted or nonsensical and when this happens the original timestamp is a sane fallback or sanity check. If you think about it in this context not maintaining the timestamp actually loses unique useful and irreplaceable data.

A real world example:

SLRs or the first decade of digital cameras relied on users to manually set the device date and time. This often did not happen resulting in crazy or no embedded date metadata e.g. epoch 0 ... 00:00:00 UTC on Thursday, 1 January 1970

**Reply by Batwam** [reply]
I agree, it would be very useful to keep the original date. I always use `rsync --times` for backups as I occasionally receive pictures without timestamps and knowing when I saved them greatly helps applying an EXIF date using that information with`exiftool`

**Comment by jextrevor** [comment]
https://github.com/immich-app/immich/pull/7010 - makes it so that the server restores the modified at timestamp after upload

**Comment by MrModest** [comment]
Just to link to similar issues: https://github.com/immich-app/immich/issues/6623
